hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpcengine/AsyncSocket_LargePayloadTest.java

```diff
@@ -214,7 +215,7 @@ public void test(int payloadSize, int concurrency) throws InterruptedException {
         }
         clientSocket.flush();
 
-        assertOpenEventually(completionLatch);
+        assertOpenEventually(completionLatch, TimeUnit.MINUTES.toSeconds(5));
```

Comment:
This test should run fast. In a few seconds it should complete.

Reply:
The integration of the timeout rule seems a bit weird. I don't see the configured timeout being fed back into the assertOpenEventually.

Reply:
> This test should run fast. In a few seconds it should complete.

In my m1 macbook pro some nightly ones take 100 secs which is concerning as 120 secs is the timeout for jenkins.

> The integration of the timeout rule seems a bit weird. I don't see the configured timeout being fed back into the assertOpenEventually.

Timeout rule is here only to override default of 5 min timeout. Since we now use 5 min latch timeout, we don't want to test itself to timeout while waiting for the latch.

https://github.com/hazelcast/hazelcast/blob/aa7c8b911a1b95aea3acf985b518ef8a4c304431/hazelcast/src/test/java/com/hazelcast/test/AbstractHazelcastClassRunner.java#L63

@TomaszGaweda IMO just remove the `Rule` and set the latch timeout to 4 minutes. Also please add a if check to only wait 4 mins if iteration count is set too high, like we do for `NioAsyncSocket_LargePayloadTest_Nightly`.

Reply:
done @ramizdundar